### PR TITLE
Recommendation

### PR DESCRIPTION
**Pull request by fredrikved**
Implement recommendations.

Continue playing from show when season ended (if next season exists)
Pick a random episode from a related collection when show ends
If no related collection, find similar episodes by tags (more common tags = more likely to appear)

### CODE REVIEW

**Comment by fredrikved** [comment]
#365

File: migrations/00183_related_collection_id.sql [hunk 1]
```diff
@@ -0,0 +1,66 @@
+-- +goose Up
```
Comment (KillerX) [inline migrations/00183_related_collection_id.sql:1]: Don't forget to validate the number before merge.

**Review by KillerX — APPROVED** [review]
A few general questions but in general it makes sense to me.

File: backend/graph/api/episode-resolver.utils.go [hunk 2]
```diff
@@ -56,17 +58,94 @@ func (r *episodeResolver) getNextEpisodes(ctx context.Context, episodeID string)
...
+	}
+	entries, err := collection.GetCollectionEntries(ctx, r.Loaders, r.GetFilteredLoaders(ctx), int(show.RelatedCollectionID.Int64))
+	if err != nil {
+		return nil, err
+	}
+	episodeEntries := lo.Filter(entries, func(i collection.Entry, _ int) bool {
+		return i.Collection == common.TypeEpisode && i.ID != episodeID
+	})
+	if len(episodeEntries) <= 0 {
+		return nil, nil
```
[the remark this answers is not in the document]
Comment (fredrikved) [reply backend/graph/api/episode-resolver.utils.go:119]: No reason to initialize an empty array. I only iterate through them, and iterating nil is the same as iterating an empty array.

**Pull request merged**